internal/cache/cleaner.go

```diff
@@ -142,12 +154,17 @@ func (c *cleaner[T, K]) Start(ctx context.Context) error {
 		}
 		objs, exp := c.collect(c.clock.Now())
 		if len(objs) > 0 {
+			c.log.Debug("cleaning up objects", "count", len(objs), "next", exp)
 			if err := c.cleanFn(objs); err != nil {
-				return err
```

Comment:
we'd want to stop the process here normally.

Reply:
fixed.

For context, it's printed here and the cache is removed:
```
	go func() {
		err := ca.Start(ctx)
		log.Debug("Cache stopped", "error", err)

		// Start blocks until ctx is closed, or it encounters an error. If we make
		// it here either the cache crashed, or the context was cancelled (e.g.
		// because our session expired).
		c.remove(id)
	}()
```
So, it's not the process in the unix sense that is stopped, but the cache.